File: register_login_api/app/Models/quiz.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Quiz extends Model
{
    protected $fillable = ['lesson_id', 'title', 'description'];

    protected $casts = [
        'choices' => 'json', // Cast 'choices' to JSON (assuming stored as JSON)
    ];
    public function lesson()
    {
        return $this->belongsTo(Lesson::class);
    }

    public function questions()
    {
        return $this->hasMany(Question::class);
    }

    public function quizScores()
    {
        return $this->hasMany(QuizScore::class);
    }   
}

File: register_login/lib/model/quiz.dart
import 'package:register_login/model/question.dart';


class Quiz {
  final int id;
  final String title;
  final List<Question> questions;

  Quiz({
    required this.id,
    required this.title,
    required this.questions,
  });

  factory Quiz.fromJson(Map<String, dynamic> json) {
    List<Question> questions = (json['questions'] as List<dynamic>)
        .map((question) => Question.fromJson(question))
        .toList();

    return Quiz(
      id: json['id'],
      title: json['title'],
      questions: questions,
    );
  }
}

